### ISSUE TITLE
Reusing single HTML2Text object on multiple documents is slow

### ISSUE THREAD

**Issue by gbrova** [Alir3z4/html2text#269]
On a corpus of ~10,000 documents, reusing the same HTML2Text object takes well over an hour:

```
  h = html2text.HTML2Text()
  h.ignore_links = True
  h.ignore_emphasis = True
  h.ignore_images = True
  
  for html in all_my_documents:
    h.handle(html)
```

But creating a new object every time takes only 2 minutes: 

```
  for html in all_my_documents:
    h = html2text.HTML2Text()
    h.ignore_links = True
    h.ignore_emphasis = True
    h.ignore_images = True
    
    h.handle(html)
```
Is there a reason why this might happen? 

I can try to construct a (smaller) corpus to reproduce this problem, if that would be helpful to debug.

Versions:
```
html2text 2018.1.9
Python 3.7.3
```

**Comment by jdufresne** [comment]
Does https://github.com/Alir3z4/html2text/commit/bd982bd31bc6ab6b4f03b5ca56b426c50c975725 help at all? Can you try the latest release?

**Comment by gbrova** [comment]
Thanks for following up, and sorry for the slowness here.  Only just had some time to check.  

Master works correctly now.  According to git bisect, the fix was actually 7e31444392d1e, which fixes a memory leak.
